web3/_utils/abi.py

```diff
@@ -163,6 +193,23 @@ def is_encodable(_type, value):
     if not isinstance(_type, str):
         raise ValueError("is_encodable only accepts type strings")
 
+    if _type[0] == "(":  # it's a tuple. check encodability of each component
```

Comment:
Yes, it is possible that the collapsing function could yield a nested tuple.  See demonstration in the tests I just added.  And you're right, `is_encodable()` does indeed fail in that case.  Let me see if I can't scheme up a reasonably elegant way to handle it.

Reply:
@davesque I believe I've addressed your concern in my latest commit, thank you for raising it.  Let me know what you think.

Reply:
@davesque would you mind taking another look?  I realize we're still deliberating strategy in #829 , but I thought it would be good to get your feedback on this while it's still relatively fresh in our minds.

For quick reference: That test showing that `collapse_if_tuple()` can indeed yield a nested tuple is [here](https://github.com/ethereum/web3.py/pull/1147/commits/506ec40568aa47ec111b99e904c370a50c6bd32d#diff-30c9fd211065dbe2e5e546e745d4f853R228).  And the change to support nested tuples in `is_encodable()`, the code in question in this thread, is [here](https://github.com/ethereum/web3.py/pull/1147/commits/5837fd608e59999f9469c6769c5092705d5a7fc7).